lib/agent.js

```diff
@@ -26,17 +26,12 @@ var version = require('../package').version
 module.exports = Agent
 
 function Agent () {
-  this.middleware = { connect: connect.bind(this) }
+  // Early configuration to ensure `agent.logger` works before `agent.start()`.
+  this.logger = config.configLogger()
 
-  this.logger = null
   this._conf = null
   this._httpClient = null
   this._uncaughtExceptionListener = null
-
-  // Early configuration to ensure `agent.logger` works before `.start()`
-  // is called.
-  this._config()
```

Comment:
*I* am. ;)

My first justification is that this is an internal/undocumented property. We cannot, IMHO, *in general* have internal code changes be disallowed by possible/unknown external usage.

The main justification is that `agent._conf` between Agent creation and `agent.start(...)` is incomplete -- a stronger word is that this _conf is *wrong* and misleading. External code should not be making behaviour decisions based on a config value that will be different after `agent.start(...)`. It could lead to a more subtle bug. (The obvious exception here is the logLevel, which we need for internal logging before `agent.start()`. If a good use case for another exception comes up, I think having the specific use case would much better inform how we support it.)

The last justification is that I would be surprised if there is any meaningful usage of the Agent instance in customer code *before* `agent.start(...)` is called.